### ISSUE TITLE
Issues to start the Container ghcr.io/linuxserver/pydio on RasperryPi (Raspbian GNU/Linux 10 (buster))

### ISSUE THREAD

**Issue by Plat0nicS0lid** [linuxserver/docker-pydio#47]
Hi, 
have some issues to start pydio8 as a docker container on RasperryPi (Raspbian GNU/Linux 10 (buster).
Run it on a standard Linux system for a while without difficulties. But to reduce some power costs, my plan is to migrate it to a Raspberry Pi. As there is no way to use pydio cells atm, I want to stay on pydio8.

I run the container with:
docker run  -d  --name=pydio   -e PUID=1000   -e PGID=1000   -e TZ=Europe/Berlin   -p 80:80  -p 443:443   -v /opt/docker/pydio/config:/config   -v /opt/docker/pydio/data:/data   --restart unless-stopped ghcr.io/linuxserver/pydio

For now, I identified three issues during the container start on my pie:
1. date provides wrong time if I switch into the bash of the pydio container after the start:
root@da2a4f4a4529:/# date
Thu Jan  1 01:00:00 CET 1970

2. I only see a cert.key file in /config/keys, but no cert.crt file
root@da2a4f4a4529:/config/keys# ls
cert.key

3. There are some errors during pydio container startup in the logs:
[cont-init.d] 30-keygen: executing...
generating self-signed keys in /config/keys, you can replace these with your own keys if required
Generating a RSA private key
...............................................................................+++++
...+++++
writing new private key to '/config/keys/cert.key'
-----
1995895696:error:0D0D90AD:asn1 encoding routines:ASN1_TIME_adj:error getting time:crypto/asn1/a_time.c:330:
[cont-init.d] 30-keygen: exited 1.
[cont-init.d] 50-install: executing...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (6) Could not resolve host: download.pydio.com
unzip:  cannot find or open install.zip, install.zip.zip or install.zip.ZIP.
mv: cannot stat 'pydio-*/data/*': No such file or directory
mv: cannot stat 'pydio-*': No such file or directory
chown: cannot access '/config/www/pydio': No such file or directory

Looks like, there are some difficulties to reach the source host download.pydio.com, maybe due to the wrong time inside the container?

Any idea why the container doesn´t start as expected? If I run the container with the same run command on a standard linux machine, it comes up as expected.

Many thanks and best regards,
Sascha

**Comment by j0nnymoe** [comment]
It should be showing you an error at the start of your logs, either way this will fix your issue: https://docs.linuxserver.io/faq#my-host-is-incompatible-with-images-based-on-ubuntu-focal

**Comment by Plat0nicS0lid** [comment]
Hi,
followed the instruction found in https://docs.linuxserver.io/faq#my-host-is-incompatible-with-images-based-on-ubuntu-focal.
But I have some doubts that this fixes the issue. Cleaned all folders, files and archives and restarted with fresh containers - same situation. The only difference is, that the libseccomp2 note is gone. But I still see the time issue. Please see logs output during container start:

_Creating pydio          ... done
Creating pydio8_mysql_1 ... done
Attaching to pydio, pydio8_mysql_1
pydio    | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
mysql_1  | 2021-03-25 18:51:46 0 [Note] mysqld (mysqld 10.3.27-MariaDB-0+deb10u1) starting as process 1 ...
mysql_1  | 2021-03-25 18:51:47 0 [Note] InnoDB: Using Linux native AIO
mysql_1  | 2021-03-25 18:51:47 0 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
mysql_1  | 2021-03-25 18:51:47 0 [Note] InnoDB: Uses event mutexes
mysql_1  | 2021-03-25 18:51:47 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
mysql_1  | 2021-03-25 18:51:47 0 [Note] InnoDB: Number of pools: 1
mysql_1  | 2021-03-25 18:51:47 0 [Note] InnoDB: Using generic crc32 instructions
pydio    | [s6-init] ensuring user provided files have correct perms...exited 0.
pydio    | [fix-attrs.d] applying ownership & permissions fixes...
pydio    | [fix-attrs.d] done.
pydio    | [cont-init.d] executing container initialization scripts...
pydio    | [cont-init.d] 01-envfile: executing...
pydio    | [cont-init.d] 01-envfile: exited 0.
pydio    | [cont-init.d] 10-adduser: executing...
mysql_1  | 2021-03-25 18:51:47 0 [Note] InnoDB: Initializing buffer pool, total size = 128M, instances = 1, chunk size = 128M
mysql_1  | 2021-03-25 18:51:47 0 [Note] InnoDB: Completed initialization of buffer pool
mysql_1  | 2021-03-25 18:51:47 0 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
mysql_1  | 2021-03-25 18:51:47 0 [Note] InnoDB: 128 out of 128 rollback segments are active.
mysql_1  | 2021-03-25 18:51:47 0 [Note] InnoDB: Creating shared tablespace for temporary tables
mysql_1  | 2021-03-25 18:51:47 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
mysql_1  | 2021-03-25 18:51:47 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
mysql_1  | 2021-03-25 18:51:47 0 [Note] InnoDB: 10.3.27 started; log sequence number 1625524; transaction id 20
mysql_1  | 2021-03-25 18:51:47 0 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
mysql_1  | 2021-03-25 18:51:47 0 [Note] Plugin 'FEEDBACK' is disabled.
mysql_1  | 2021-03-25 18:51:47 0 [Note] InnoDB: Buffer pool(s) load completed at 210325 18:51:47
mysql_1  | 2021-03-25 18:51:47 0 [Note] Server socket created on IP: '::'.
mysql_1  | 2021-03-25 18:51:47 0 [Warning] 'proxies_priv' entry '@% root@42df393857cd' ignored in --skip-name-resolve mode.
mysql_1  | 2021-03-25 18:51:47 0 [Note] Reading of all Master_info entries succeeded
mysql_1  | 2021-03-25 18:51:47 0 [Note] Added new Master_info '' to hash table
mysql_1  | 2021-03-25 18:51:47 0 [Note] mysqld: ready for connections.
mysql_1  | Version: '10.3.27-MariaDB-0+deb10u1'  socket: '/run/mysqld/mysqld.sock'  port: 3306  Raspbian 10
pydio    |
pydio    | -------------------------------------
pydio    |           _         ()
pydio    |          | |  ___   _    __
pydio    |          | | / __| | |  /  \
pydio    |          | | \__ \ | | | () |
pydio    |          |_| |___/ |_|  \__/
pydio    |
pydio    |
pydio    | Brought to you by linuxserver.io
pydio    | -------------------------------------
pydio    |
pydio    | To support LSIO projects visit:
pydio    | https://www.linuxserver.io/donate/
pydio    | -------------------------------------
pydio    | GID/UID
pydio    | -------------------------------------
pydio    |
pydio    | User uid:    1000
pydio    | User gid:    1000
pydio    | -------------------------------------
pydio    |
pydio    |
pydio    | @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
pydio    |
pydio    | Your DockerHost is most likely running an outdated version of libseccomp
pydio    |
pydio    | To fix this, please visit https://docs.linuxserver.io/faq#libseccomp
pydio    |
pydio    | Some apps might not behave correctly without this
pydio    |
pydio    | @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
pydio    |
pydio    | [cont-init.d] 10-adduser: exited 0.
pydio    | [cont-init.d] 20-config: executing...
pydio    | [cont-init.d] 20-config: exited 0.
pydio    | [cont-init.d] 30-keygen: executing...
pydio    | generating self-signed keys in /config/keys, you can replace these with your own keys if required
pydio    | Generating a RSA private key
pydio    | ..........................................................................+++++
pydio    | .............................................................................................................+++++
pydio    | writing new private key to '/config/keys/cert.key'
pydio    | -----
pydio    | 1996428176:error:0D0D90AD:asn1 encoding routines:ASN1_TIME_adj:error getting time:crypto/asn1/a_time.c:330:
pydio    | [cont-init.d] 30-keygen: exited 1.
pydio    | [cont-init.d] 50-install: executing...
pydio    |   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
pydio    |                                  Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (6) Could not resolve host: download.pydio.com
pydio    | unzip:  cannot find or open install.zip, install.zip.zip or install.zip.ZIP.
pydio    | mv: cannot stat 'pydio-*/data/*': No such file or directory
pydio    | mv: cannot stat 'pydio-*': No such file or directory
pydio    | chown: cannot access '/config/www/pydio': No such file or directory
pydio    | [cont-init.d] 50-install: exited 1.
pydio    | [cont-init.d] 60-config: executing...
pydio    | sed: /config/www/pydio/conf/bootstrap_context.php: No such file or directory
pydio    | [cont-init.d] 60-config: exited 0.
pydio    | [cont-init.d] 99-custom-files: executing...
pydio    | [custom-init] no custom files found exiting...
pydio    | [cont-init.d] 99-custom-files: exited 0.
pydio    | [cont-init.d] done.
pydio    | [services.d] starting services
pydio    | [services.d] done.
pydio    | nginx: [warn] the "ssl" directive is deprecated, use the "listen ... ssl" directive instead in /config/nginx/site-confs/default:8
pydio    | nginx: [emerg] cannot load certificate "/config/keys/cert.crt": BIO_new_file() failed (SSL: error:02001002:system library:fopen:No such file or directory:fopen('/config/keys/cert.crt','r') error:2006D080:BIO routines:BIO_new_file:no such file)
pydio    | nginx: [warn] the "ssl" directive is deprecated, use the "listen ... ssl" directive instead in /config/nginx/site-confs/default:8
pydio    | nginx: [emerg] cannot load certificate "/config/keys/cert.crt": BIO_new_file() failed (SSL: error:02001002:system library:fopen:No such file or directory:fopen('/config/keys/cert.crt','r') error:2006D080:BIO routines:BIO_new_file:no such file)
pydio    | nginx: [warn] the "ssl" directive is deprecated, use the "listen ... ssl" directive instead in /config/nginx/site-confs/default:8
pydio    | nginx: [emerg] cannot load certificate "/config/keys/cert.crt": BIO_new_file() failed (SSL: error:02001002:system library:fopen:No such file or directory:fopen('/config/keys/cert.crt','r') error:2006D080:BIO routines:BIO_new_file:no such file)
pydio    | nginx: [warn] the "ssl" directive is deprecated, use the "listen ... ssl" directive instead in /config/nginx/site-confs/default:8
pydio    | nginx: [emerg] cannot load certificate "/config/keys/cert.crt": BIO_new_file() failed (SSL: error:02001002:system library:fopen:No such file or directory:fopen('/config/keys/cert.crt','r') error:2006D080:BIO routines:BIO_new_file:no such file)
pydio    | nginx: [warn] the "ssl" directive is deprecated, use the "listen ... ssl" directive instead in /config/nginx/site-confs/default:8
pydio    | nginx: [emerg] cannot load certificate "/config/keys/cert.crt": BIO_new_file() failed (SSL: error:02001002:system library:fopen:No such file or directory:fopen('/config/keys/cert.crt','r') error:2006D080:BIO routines:BIO_new_file:no such file)_

Best regards,
Sascha

**Comment by Roxedus** [comment]
Your log indicate that the fix from the FAQ was not applied.

**Comment by Plat0nicS0lid** [comment]
Yeah, become aware of it right now as well. Unfortunately, I was not fast enough to delete my reply again.

**Comment by Plat0nicS0lid** [comment]
Finally, I found the issue(s). On one hand it was that libseccomp2 situation, but on the other hand there was another self-created issue preventing to let curl download the file.

Remember that piece:
_% Total % Received % Xferd Average Speed Time Time Time Current
Dload Upload Total Spent Left Speed
0 0 0 0 0 0 0 0 --:--:-- --:--:-- --:--:-- 0curl: (6) Could not resolve host: download.pydio.com_
The reason was a typo in my network section in my docker_compose.yaml file.
After used the default bridge, the access/ping to download.pydio.com worked. So my network configuration was the second thing.

Regarding the libseccomp2 thing, I downloaded the 2.5.1.1-1 version for armhf.
_wget http://ftp.us.debian.org/debian/pool/main/libs/libseccomp/libseccomp2_2.5.1-1_armhf.deb
dpkg -i libseccomp2_2.5.1-1_armhf.deb_
Error message gone and the second cert file exists now.

Many thanks for the guidance and best regards,
Sascha
